rust.hcl

```diff
@@ -6,14 +6,20 @@ env = {
   "PATH": "${HERMIT_ENV}/.hermit/rust/bin:${PATH}",
 }
 
-darwin {
-  sha256-source = "https://static.rust-lang.org/dist/rust-${version}-${xarch}-apple-darwin.tar.xz.sha256"
-  source = "https://static.rust-lang.org/dist/rust-${version}-${xarch}-apple-darwin.tar.xz"
+platform "darwin" "amd64" {
+  source = "https://static.rust-lang.org/dist/rust-${version}-x86_64-apple-darwin.tar.xz"
 }
 
-linux {
-  sha256-source = "https://static.rust-lang.org/dist/rust-${version}-${xarch}-unknown-linux-gnu.tar.xz.sha256"
-  source = "https://static.rust-lang.org/dist/rust-${version}-${xarch}-unknown-linux-gnu.tar.xz"
+platform "darwin" "arm64" {
+  source = "https://static.rust-lang.org/dist/rust-${version}-aarch64-apple-darwin.tar.xz"
+}
+
+platform "linux" "amd64" {
+  source = "https://static.rust-lang.org/dist/rust-${version}-x86_64-unknown-linux-gnu.tar.xz"
+}
+
+platform "linux" "arm64" {
+  source = "https://static.rust-lang.org/dist/rust-${version}-aarch64-unknown-linux-gnu.tar.xz"
 }
```

Comment:
I think this can be simplified down to just:

```
darwin {
  source = "https://static.rust-lang.org/dist/rust-${version}-${xarch}-apple-darwin.tar.xz"
}

linux {
  source = "https://static.rust-lang.org/dist/rust-${version}-${xarch}-unknown-linux-gnu.tar.xz"
}
```

It's unclear why to me at this stage, but the error only seems to trigger when the `sha256-source` field tries to use the `version` variable.